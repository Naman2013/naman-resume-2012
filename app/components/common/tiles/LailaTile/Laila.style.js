import css from 'styled-jsx/css';
import { primaryFont, secondaryFont } from 'app/styles/variables/fonts';
import {
  white_tile,
  astronaut,
  blue_tile_canvas,
} from 'app/styles/variables/colors_tiles_v4';
import { screenMedium } from 'app/styles/variables/breakpoints';

export default css`
  .root {
    width: 100%;
    height: 142px;
    box-sizing: border-box;
    padding: 0;
    box-shadow: 0 0 8px 1px rgba(65, 86, 113, 0.2);
    color: ${astronaut};
    list-style-type: none;
    background: url(${white_tile});
    background-size: 30px;
    transition: width, height 0.15s ease-out;
  }

  .container {
    width: 80%;
    margin: 0 auto;
  }

  .icon-container {
    display: none;
  }

  .center-line {
    border-right: 1px solid #d9dee4;
    width: 50%;
    height: 60px;
  }

  .border {
    width: 115px;
    height: 115px;
    border: 1px dashed #d9dee4;
    border-radius: 50%;
    margin: 0 auto;
    position: relative;
  }

  .icon {
    position: absolute;
    width: 96px;
    height: 96px;
    margin: 0 auto;
    border-radius: 50%;
    background: url(${blue_tile_canvas});
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  .icon-content {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  .title {
    padding: 30px 0 20px 0;
    margin: 0;
    font-family: ${secondaryFont};
    font-weight: 300;
    font-size: 18px;
    border-bottom: 1px solid #d9dee4;
  }

  .button-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-top: 20px;
  }

  .button-container :global(a) {
    display: block;
    padding: 0;
    color: ${astronaut};
    text-decoration: none;
    text-transform: uppercase;
    font-family: ${primaryFont};
    font-size: 10px;
    font-weight: bold;
    letter-spacing: 2px;
  }

  @media ${screenMedium} {
    .root {
      width: 300px;
      height: 332px;
      text-align: center;
    }

    .icon-container {
      display: block;
    }

    .title {
      border-bottom: none;
      padding-top: 50px;
      padding-bottom: 0;
    }

    .button-container {
      justify-content: center;
    }

    .button-container :global(a) {
      padding-right: 10px;
    }
  }
`;
