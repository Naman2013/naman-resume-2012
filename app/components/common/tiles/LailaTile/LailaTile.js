import React from 'react';
import PropTypes from 'prop-types';
import style from './Laila.style';
import { horizontalArrow } from 'styles/variables/iconURLs';

const LailaTile = ({ iconURL, title }) => (
  <div className="root">
    <div className="container">

      <div className="icon-container">
        <div className="center-line" />
        <div className="border">
          <div className="icon">
            <img className="icon-content" alt="" width="40" height="40" src={iconURL} />
          </div>
        </div>
      </div>

      <h5 className="title">{title}</h5>
      <div className="button-container">
        <a className="button" href="#">Explore now</a>
        <img alt="" src={horizontalArrow} />
      </div>
    </div>
    <style jsx>{style}</style>
  </div>
);

LailaTile.propTypes = {
  iconURL: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default LailaTile;
