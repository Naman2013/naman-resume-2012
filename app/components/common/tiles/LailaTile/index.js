import LailaTile from 'LailaTile';

export default LailaTile;
