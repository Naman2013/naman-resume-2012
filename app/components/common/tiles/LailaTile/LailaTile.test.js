import React from 'react';
import { shallow } from 'enzyme';
import LailaTile from './LailaTile';

describe('LailaTile', () => {
  const shallowWrapper = shallow(<LailaTile />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
