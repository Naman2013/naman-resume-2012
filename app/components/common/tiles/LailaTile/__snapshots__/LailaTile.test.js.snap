// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LailaTile should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LailaTile
    iconURL="https://vega.slooh.com/icons/home/jupiter-icon.png"
    title="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="jsx-1632977626 container"
        >
          <div
            className="jsx-1632977626 icon-container"
          >
            <div
              className="jsx-1632977626 center-line"
            />
            <div
              className="jsx-1632977626 border"
            >
              <div
                className="jsx-1632977626 icon"
              >
                <img
                  alt=""
                  className="jsx-1632977626 icon-content"
                  height="40"
                  src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                  width="40"
                />
              </div>
            </div>
          </div>
          <h5
            className="jsx-1632977626 title"
          >
            Test
          </h5>
          <a
            className="jsx-1632977626 button"
            href="#"
          >
            Explore now
          </a>
        </div>,
        <JSXStyle
          css={
            Array [
              ".root.jsx-1632977626{width:300px;height:142px;box-size:border-box;padding:0;box-shadow:0px 0px 15px 1px rgba(65,86,113,.2);color:#415671;list-style-type:none;background:url(https://vega.slooh.com/assets/v4/common/LightBlueDiamondPattern_NO_TEXTURE.png);background-size:30px;-webkit-transition:width,height 0.15s ease-out;transition:width,height 0.15s ease-out;}",
              ".container.jsx-1632977626{width:80%;margin:0 auto;}",
              ".icon-container.jsx-1632977626{display:none;}",
              ".center-line.jsx-1632977626{border-right:1px solid #D9DEE4;width:50%;height:60px;}",
              ".border.jsx-1632977626{width:115px;height:115px;border:1px dashed #D9DEE4;border-radius:50%;margin:0 auto;position:relative;}",
              ".icon.jsx-1632977626{position:absolute;width:100px;height:100px;margin:0 auto;border-radius:50%;background:url(https://vega.slooh.com/assets/v4/common/Darkest_Pattern_Canvas.png);top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
              ".icon-content.jsx-1632977626{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
              ".title.jsx-1632977626{padding:30px 0 20px 0;margin:0;font-size:14px;font-family:adobe-garamond-pro, serif;font-weight:300;font-size:18px;border-bottom:1px solid #D9DEE4;}",
              ".button.jsx-1632977626{display:block;padding:0;padding-top:20px;color:#415671;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;font-family:brandon-grotesque, sans-serif;font-size:10px;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;}",
              "@media screen and (min-width: 768px){.root.jsx-1632977626{height:332px;text-align:center;}.icon-container.jsx-1632977626{display:block;}.title.jsx-1632977626{border-bottom:none;padding-top:50px;padding-bottom:0;}}",
            ]
          }
          styleId="1632977626"
        />,
      ],
      "className": "jsx-1632977626 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="jsx-1632977626 icon-container"
            >
              <div
                className="jsx-1632977626 center-line"
              />
              <div
                className="jsx-1632977626 border"
              >
                <div
                  className="jsx-1632977626 icon"
                >
                  <img
                    alt=""
                    className="jsx-1632977626 icon-content"
                    height="40"
                    src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                    width="40"
                  />
                </div>
              </div>
            </div>,
            <h5
              className="jsx-1632977626 title"
            >
              Test
            </h5>,
            <a
              className="jsx-1632977626 button"
              href="#"
            >
              Explore now
            </a>,
          ],
          "className": "jsx-1632977626 container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="jsx-1632977626 center-line"
                />,
                <div
                  className="jsx-1632977626 border"
                >
                  <div
                    className="jsx-1632977626 icon"
                  >
                    <img
                      alt=""
                      className="jsx-1632977626 icon-content"
                      height="40"
                      src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                      width="40"
                    />
                  </div>
                </div>,
              ],
              "className": "jsx-1632977626 icon-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "jsx-1632977626 center-line",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="jsx-1632977626 icon"
                  >
                    <img
                      alt=""
                      className="jsx-1632977626 icon-content"
                      height="40"
                      src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                      width="40"
                    />
                  </div>,
                  "className": "jsx-1632977626 border",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt=""
                      className="jsx-1632977626 icon-content"
                      height="40"
                      src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                      width="40"
                    />,
                    "className": "jsx-1632977626 icon",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "className": "jsx-1632977626 icon-content",
                      "height": "40",
                      "src": "https://vega.slooh.com/icons/home/jupiter-icon.png",
                      "width": "40",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "jsx-1632977626 title",
            },
            "ref": null,
            "rendered": "Test",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Explore now",
              "className": "jsx-1632977626 button",
              "href": "#",
            },
            "ref": null,
            "rendered": "Explore now",
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".root.jsx-1632977626{width:300px;height:142px;box-size:border-box;padding:0;box-shadow:0px 0px 15px 1px rgba(65,86,113,.2);color:#415671;list-style-type:none;background:url(https://vega.slooh.com/assets/v4/common/LightBlueDiamondPattern_NO_TEXTURE.png);background-size:30px;-webkit-transition:width,height 0.15s ease-out;transition:width,height 0.15s ease-out;}",
            ".container.jsx-1632977626{width:80%;margin:0 auto;}",
            ".icon-container.jsx-1632977626{display:none;}",
            ".center-line.jsx-1632977626{border-right:1px solid #D9DEE4;width:50%;height:60px;}",
            ".border.jsx-1632977626{width:115px;height:115px;border:1px dashed #D9DEE4;border-radius:50%;margin:0 auto;position:relative;}",
            ".icon.jsx-1632977626{position:absolute;width:100px;height:100px;margin:0 auto;border-radius:50%;background:url(https://vega.slooh.com/assets/v4/common/Darkest_Pattern_Canvas.png);top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
            ".icon-content.jsx-1632977626{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
            ".title.jsx-1632977626{padding:30px 0 20px 0;margin:0;font-size:14px;font-family:adobe-garamond-pro, serif;font-weight:300;font-size:18px;border-bottom:1px solid #D9DEE4;}",
            ".button.jsx-1632977626{display:block;padding:0;padding-top:20px;color:#415671;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;font-family:brandon-grotesque, sans-serif;font-size:10px;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;}",
            "@media screen and (min-width: 768px){.root.jsx-1632977626{height:332px;text-align:center;}.icon-container.jsx-1632977626{display:block;}.title.jsx-1632977626{border-bottom:none;padding-top:50px;padding-bottom:0;}}",
          ],
          "styleId": "1632977626",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="jsx-1632977626 container"
          >
            <div
              className="jsx-1632977626 icon-container"
            >
              <div
                className="jsx-1632977626 center-line"
              />
              <div
                className="jsx-1632977626 border"
              >
                <div
                  className="jsx-1632977626 icon"
                >
                  <img
                    alt=""
                    className="jsx-1632977626 icon-content"
                    height="40"
                    src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                    width="40"
                  />
                </div>
              </div>
            </div>
            <h5
              className="jsx-1632977626 title"
            >
              Test
            </h5>
            <a
              className="jsx-1632977626 button"
              href="#"
            >
              Explore now
            </a>
          </div>,
          <JSXStyle
            css={
              Array [
                ".root.jsx-1632977626{width:300px;height:142px;box-size:border-box;padding:0;box-shadow:0px 0px 15px 1px rgba(65,86,113,.2);color:#415671;list-style-type:none;background:url(https://vega.slooh.com/assets/v4/common/LightBlueDiamondPattern_NO_TEXTURE.png);background-size:30px;-webkit-transition:width,height 0.15s ease-out;transition:width,height 0.15s ease-out;}",
                ".container.jsx-1632977626{width:80%;margin:0 auto;}",
                ".icon-container.jsx-1632977626{display:none;}",
                ".center-line.jsx-1632977626{border-right:1px solid #D9DEE4;width:50%;height:60px;}",
                ".border.jsx-1632977626{width:115px;height:115px;border:1px dashed #D9DEE4;border-radius:50%;margin:0 auto;position:relative;}",
                ".icon.jsx-1632977626{position:absolute;width:100px;height:100px;margin:0 auto;border-radius:50%;background:url(https://vega.slooh.com/assets/v4/common/Darkest_Pattern_Canvas.png);top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
                ".icon-content.jsx-1632977626{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
                ".title.jsx-1632977626{padding:30px 0 20px 0;margin:0;font-size:14px;font-family:adobe-garamond-pro, serif;font-weight:300;font-size:18px;border-bottom:1px solid #D9DEE4;}",
                ".button.jsx-1632977626{display:block;padding:0;padding-top:20px;color:#415671;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;font-family:brandon-grotesque, sans-serif;font-size:10px;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;}",
                "@media screen and (min-width: 768px){.root.jsx-1632977626{height:332px;text-align:center;}.icon-container.jsx-1632977626{display:block;}.title.jsx-1632977626{border-bottom:none;padding-top:50px;padding-bottom:0;}}",
              ]
            }
            styleId="1632977626"
          />,
        ],
        "className": "jsx-1632977626 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="jsx-1632977626 icon-container"
              >
                <div
                  className="jsx-1632977626 center-line"
                />
                <div
                  className="jsx-1632977626 border"
                >
                  <div
                    className="jsx-1632977626 icon"
                  >
                    <img
                      alt=""
                      className="jsx-1632977626 icon-content"
                      height="40"
                      src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                      width="40"
                    />
                  </div>
                </div>
              </div>,
              <h5
                className="jsx-1632977626 title"
              >
                Test
              </h5>,
              <a
                className="jsx-1632977626 button"
                href="#"
              >
                Explore now
              </a>,
            ],
            "className": "jsx-1632977626 container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="jsx-1632977626 center-line"
                  />,
                  <div
                    className="jsx-1632977626 border"
                  >
                    <div
                      className="jsx-1632977626 icon"
                    >
                      <img
                        alt=""
                        className="jsx-1632977626 icon-content"
                        height="40"
                        src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                        width="40"
                      />
                    </div>
                  </div>,
                ],
                "className": "jsx-1632977626 icon-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "jsx-1632977626 center-line",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="jsx-1632977626 icon"
                    >
                      <img
                        alt=""
                        className="jsx-1632977626 icon-content"
                        height="40"
                        src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                        width="40"
                      />
                    </div>,
                    "className": "jsx-1632977626 border",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <img
                        alt=""
                        className="jsx-1632977626 icon-content"
                        height="40"
                        src="https://vega.slooh.com/icons/home/jupiter-icon.png"
                        width="40"
                      />,
                      "className": "jsx-1632977626 icon",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "",
                        "className": "jsx-1632977626 icon-content",
                        "height": "40",
                        "src": "https://vega.slooh.com/icons/home/jupiter-icon.png",
                        "width": "40",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "jsx-1632977626 title",
              },
              "ref": null,
              "rendered": "Test",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Explore now",
                "className": "jsx-1632977626 button",
                "href": "#",
              },
              "ref": null,
              "rendered": "Explore now",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".root.jsx-1632977626{width:300px;height:142px;box-size:border-box;padding:0;box-shadow:0px 0px 15px 1px rgba(65,86,113,.2);color:#415671;list-style-type:none;background:url(https://vega.slooh.com/assets/v4/common/LightBlueDiamondPattern_NO_TEXTURE.png);background-size:30px;-webkit-transition:width,height 0.15s ease-out;transition:width,height 0.15s ease-out;}",
              ".container.jsx-1632977626{width:80%;margin:0 auto;}",
              ".icon-container.jsx-1632977626{display:none;}",
              ".center-line.jsx-1632977626{border-right:1px solid #D9DEE4;width:50%;height:60px;}",
              ".border.jsx-1632977626{width:115px;height:115px;border:1px dashed #D9DEE4;border-radius:50%;margin:0 auto;position:relative;}",
              ".icon.jsx-1632977626{position:absolute;width:100px;height:100px;margin:0 auto;border-radius:50%;background:url(https://vega.slooh.com/assets/v4/common/Darkest_Pattern_Canvas.png);top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
              ".icon-content.jsx-1632977626{position:absolute;top:50%;left:50%;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
              ".title.jsx-1632977626{padding:30px 0 20px 0;margin:0;font-size:14px;font-family:adobe-garamond-pro, serif;font-weight:300;font-size:18px;border-bottom:1px solid #D9DEE4;}",
              ".button.jsx-1632977626{display:block;padding:0;padding-top:20px;color:#415671;-webkit-text-decoration:none;text-decoration:none;text-transform:uppercase;font-family:brandon-grotesque, sans-serif;font-size:10px;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;}",
              "@media screen and (min-width: 768px){.root.jsx-1632977626{height:332px;text-align:center;}.icon-container.jsx-1632977626{display:block;}.title.jsx-1632977626{border-bottom:none;padding-top:50px;padding-bottom:0;}}",
            ],
            "styleId": "1632977626",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
